### PR TITLE
add d3-sankey@0.4.1 w/ npm auto-update

### LINKED ISSUE #9983: [Request] Add sankey.js
**Library name:** sankey.js
**Git repository url:** https://github.com/d3/d3-plugins/tree/master/sankey
**npm package url(optional):** 
**License(s):**
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #9983
Related issue(s): # #
@kennynaoh please help me review this PR, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/d3/d3-sankey
 * Official website (optional, not the repository): https://bost.ocks.org/mike/sankey/
 * NPM package url (optional): https://www.npmjs.com/package/d3-sankey
 * License and its reference: BSD-3-Clause
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 12
   - Count of stars: 54
   - Count of forks: 20
 * NPM download stats (optional):
   - Downloads in the last day: 31
   - Downloads in the last week: 553
   - Downloads in the last month: 1916

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/d3-sankey/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "name": "d3-sankey",
+  "version": "0.4.1",
+  "description": "Visualize flow between nodes in a directed acyclic network.",
+  "keywords": [
+    "d3",
+    "sankey"
+  ],
+  "author": {
+    "name": "Mike Bostock",
+    "url": "http://bost.ocks.org/mike"
```
Comment (maruilian11) [inline ajax/libs/d3-sankey/package.json]: Please also use `https`, too.

File: ajax/libs/d3-sankey/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "license": "BSD-3-Clause",
+  "homepage": "https://bost.ocks.org/mike/sankey/",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/d3/d3-sankey.git"
+  },
+  "filename": "d3-sankey.min.js",
```
Comment (maruilian11) [inline ajax/libs/d3-sankey/package.json]: Putting `filename` in previous field is better.

**Comment by maruilian11** [comment]
please do not mentsion `close issue and cc the author` in comments next time.

**Comment by sufuf3** [comment]
@maruilian11 thanks for your review.
I modified it, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
this config will fetch `bundle.js` that we don't need, please add those version manually, thanks!

**Comment by sufuf3** [comment]
@maruilian11 I modified the package.json, and it can be fit the other version, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
We prefer use wild regular expression in config when meeting `dist` or `build` folder. I think add those two old version manually is better.

**Comment by sufuf3** [comment]
@maruilian11  thanks for your suggest.
I add the old versions and modified the package.json file, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
we add old version manually because we don't want to fetch `bundle.js`.
and please add adding old versions as another commit.

**Comment by sufuf3** [comment]
@maruilian11 thanks for your review.
I modified it, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
You still add `bundle.js` that we don't need. Do you know why we add them manually it? IF you don't see it, why not asking?

**Comment by sufuf3** [comment]
@maruilian11  sorry about that.
I remove `bundle.js`, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
please provide the source reference of those old versions.

**Comment by sufuf3** [comment]
@maruilian11 the old versions add from npm, I show the description in commit.

**Comment by maruilian11** [comment]
@sufuf3 I think this PR is ok, please rebase this branch with the latest master.

**Comment by sufuf3** [comment]
I rebase this branch with the latest master.

**Comment by kennynaoh** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
@kennynaoh use "approve" feature next time, thanks.

**Pull request merged**